bundles/org.openhab.binding.dominoswiss/src/main/java/org/openhab/binding/dominoswiss/internal/eGateHandler.java

```diff
@@ -0,0 +1,344 @@
+        logger.debug("EGate got command: {}", command);
+
+        if (!isConnected()) {
+            logger.debug("no connection to Dominoswiss eGate server when trying to send command, returning...");
+            return;
+        }
+
+        // Send plain string to eGate Server,
+        logger.debug("Sending command: {}", command);
+        try {
+
+            // BufferedWriter writer = new BufferedWriter(new OutputStreamWriter(egateSocket.getOutputStream()));
+
+            egateSocket.setSoTimeout(SOCKET_TIMEOUT);
+            writer.write(command);
+            writer.flush();
+            logger.debug("Sent message");
+        } catch (IOException e) {
+            logger.error("Error while sending command {} to Dominoswiss eGate Server {} ", command, e.toString());
+        }
+    }
+
+    private void startAutomaticRefresh() {
+        Runnable runnable = new Runnable() {
+            @SuppressWarnings("null")
+            @Override
+            public void run() {
+                try {
+                    if (reader == null) {
+                        reader = new BufferedReader(new InputStreamReader(egateSocket.getInputStream()));
+                    }
+                    logger.debug("Socket State: {} to: {}", egateSocket.isConnected(), egateSocket.toString());
+                    if (reader.ready()) {
+                        String input = reader.readLine();
+                        logger.debug("Reader got from EGATE: {}", input);
+                        onData(input);
+                    }
+                } catch (IOException e) {
+                    logger.error("Error while reading command from Dominoswiss eGate Server {} ", e.toString());
+                    updateStatus(ThingStatus.OFFLINE);
+                }
+            }
+        };
+        refreshJob = scheduler.scheduleWithFixedDelay(runnable, 0, 1, TimeUnit.SECONDS);
+    }
+
+    /**
+     * Finds and returns a child thing for a given UID of this bridge.
+     *
+     * @param uid uid of the child thing
+     * @return child thing with the given uid or null if thing was not found
+     */
+    public @Nullable Thing getThingByUID(ThingUID uid) {
+        Bridge bridge = getThing();
+
+        List<Thing> things = bridge.getThings();
+
+        for (Thing thing : things) {
+            if (thing.getUID().equals(uid)) {
+                return thing;
+            }
+        }
+
+        return null;
+    }
+
+    @SuppressWarnings("null")
+    protected void onData(String input) {
+
+        // Instruction=2;ID=19;Command=1;Value=0;Priority=0;
+
+        Map<String, String> map = new HashMap<String, String>();
+        Thing blind;
+        // split on ;
+        String[] parts = input.split(";");
+
+        for (int i = 0; i < parts.length; i += 2) {
+            map.put(parts[i], parts[i + 1]);
+        }
+
+        // only use FSSReceive Commands
+        if (map.get("Instruction").toString() == "2") {
```

Comment:
This will crash if the remote didn't sent an Instruction key. Why not integrating this code into the for loop? Then, you don't need a Map either.